nav2_bt_navigator/src/bt_navigator.cpp

```diff
@@ -249,14 +272,15 @@ BtNavigator::navigateToPose()
       return action_server_->is_cancel_requested();
     };
 
-  auto bt_xml_filename = action_server_->get_current_goal()->behavior_tree;
+  std::string bt_xml_filename = action_server_->get_current_goal()->behavior_tree.c_str();
```

Comment:
Revert this, L279 and L283

Reply:
I had some nasty errors with auto in this particular section. I currently am not able to remember what was the concrete issue here. But I believe that it was not detected that a tree was switched. With changing it to the current state, I can reliable detect different xml_file_paths send through the action_goal.

But if you want I can investigate further why there was an error and solve this with auto.
I need to properly test this section again.